Pear/Controller/SchedulingLocationCollectionViewController.swift

```diff
@@ -0,0 +1,323 @@
+        )
+        collectionView.register(SchedulingPlaceCollectionViewCell.self, forCellWithReuseIdentifier: SchedulingPlaceCollectionViewCell.onlineReuseId)
+        collectionView.register(LocationHeaderLabelView.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: onlineHeaderIdentifier)
+        collectionView.register(LocationFooterLabelView.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionFooter, withReuseIdentifier: onlineFooterIdentifer)
+
+        collectionView.snp.makeConstraints { make in
+            make.leading.trailing.top.bottom.equalToSuperview()
+        }
+
+        setupLocationSections()
+    }
+
+    // MARK: - Setup Functions
+    private func setupLocationSections() {
+        if let isChoosing = isChoosing, isChoosing {
+            campusLocations = campusLocations.filter(matchLocations.contains)
+            ctownLocations = ctownLocations.filter(matchLocations.contains)
+            onlineLocations = onlineLocations.filter(matchLocations.contains)
+        }
+        locationSections = [
+            .online(onlineLocations),
+            .campus(campusLocations),
+            .ctown(ctownLocations)
+        ]
+        if schedulingStatus == .confirming {
+            preselectSavedLocations()
+        }
+    }
+
+    private func preselectSavedLocations() {
+        selectedCampusLocations = campusLocations.filter { savedLocations.contains($0) }
+        selectedCtownLocations = ctownLocations.filter { savedLocations.contains($0) }
+        selectedOnlineLocations = onlineLocations.filter { savedLocations.contains($0) }
+
+        for onlineLocation in selectedOnlineLocations {
+            guard let index = onlineLocations.firstIndex(of: onlineLocation) else { continue }
+            collectionView.selectItem(at: IndexPath(item: index, section: 0), animated: false, scrollPosition: .top)
+        }
+
+        for campusLocation in selectedCampusLocations {
+            guard let index = campusLocations.firstIndex(of: campusLocation) else { continue }
+            collectionView.selectItem(at: IndexPath(item: index, section: 0), animated: false, scrollPosition: .top)
+        }
+        for ctownLocation in selectedCtownLocations {
+            guard let index = ctownLocations.firstIndex(of: ctownLocation) else { continue }
+            collectionView.selectItem(at: IndexPath(item: index, section: 1), animated: false, scrollPosition: .top)
+        }
+
+        if let updateNext = updateNext {
+            updateNext(selectedCampusLocations.count + selectedCtownLocations.count)
+        }
+    }
+
+
+    // MARK: - UICollectionViewDelegate
+    override func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
+        totalSelectedLocations < 3
+    }
+
+    override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let selectedLocation: String
```

Comment:
There's no point in putting this up here. You only read and write to it within the scope of each switch case. So delete this line, delete 188, and replace `selectedLocation` on 190 with `locations[indexPath.row]`. Similar thing in the other cases.